src/main/java/com/dell/isg/smi/wsman/command/CreateTargetedConfigJob.java

```diff
@@ -0,0 +1,206 @@
+
+import com.dell.isg.smi.wsman.WSCommandRNDConstant;
+import com.dell.isg.smi.wsman.WSManBaseCommand;
+import com.dell.isg.smi.wsman.WSManageSession;
+import com.sun.ws.management.addressing.Addressing;
+
+/**
+ * @author Muqeeth.Kowkab
+ *
+ */
+public class CreateTargetedConfigJob extends WSManBaseCommand {
+	
+	    private static final Logger logger = LoggerFactory.getLogger(CreateTargetedConfigJob.class);
+	    
+	    private WSManageSession session = null;
+
+		private Map<String, String> result = new HashMap<String, String>();
+
+	public CreateTargetedConfigJob(String ipAddr, String userName, String passwd, String target, int rebootJobType,
+			String scheduledStartTime, String untilTime) {		
+		    super(ipAddr, userName, passwd);
+	       
+	        session = super.getSession();
+	        session.setResourceUri(getResourceURI(target));
+	        this.addSelectors(getResourceURI(target));
+
+	        session.setInvokeCommand(WSManMethodEnum.CREATE_TARGET_CONFIG_JOB.toString());
+	        if (rebootJobType <=0 || rebootJobType >3) {
+	        	session.addUserParam(WSManMethodParamEnum.REBOOT_JOB_TYPE.toString(), Integer.toString(rebootJobType));
+	        } else {
+	        	session.addUserParam(WSManMethodParamEnum.REBOOT_JOB_TYPE.toString(), "3");
+	        }      
+
+	        session.addUserParam(WSManMethodParamEnum.TARGET.toString(), target);
+	        session.addUserParam(WSManMethodParamEnum.SCHEDULED_START_TIME.toString(), scheduledStartTime);
+	        session.addUserParam(WSManMethodParamEnum.UNTIL_TIME.toString(), untilTime);
+
+	        logger.trace("Exiting constructor: CreateTargetedConfigJob()");
+	}
+
+	private String getResourceURI(String target) {
+		StringBuilder sb = new StringBuilder(WSCommandRNDConstant.WSMAN_BASE_URI);
+		if (StringUtils.equals(target, "System.Embedded.1")) {
+	    	sb.append(WSCommandRNDConstant.WS_OS_SVC_NAMESPACE).append(WSManClassEnum.DCIM_SystemManagementService);
+	    } else if (StringUtils.equals(target, "BIOS.Setup.1-1")) {
+	    	sb.append(WSCommandRNDConstant.WS_OS_SVC_NAMESPACE).append(WSManClassEnum.DCIM_BIOSService);
+	    }    
+
+	    return sb.toString();
+	}
+
+	@Override
+	public Map<String, String> execute() throws Exception {
+		long interval = 30;
+		TimeUnit unit = TimeUnit.SECONDS;
+		try {
+			final TargetConfigJob targetConfigJob = new TargetConfigJob();
+			targetConfigJob.configJob();
+			
+			if (!result.containsKey("JobId")) {
+				final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+								
+				for (int i=0; i<5; i++) {					
+					ScheduledFuture<Map<String, String>> future = scheduler.schedule(targetConfigJob, interval, unit);
+					Map<String, String> output = future.get();
+					if (StringUtils.isNotBlank(output.get("JobId"))) {						
+						future.cancel(true);
+						return result;
+					}
+					
+				}
+				scheduler.shutdown();
+				try {
+					if (!scheduler.awaitTermination(1000, TimeUnit.MILLISECONDS)) {
```

Comment:
Please move this into exception block or after exception to properly shutdown the scheduler